sympy/solvers/solvers.py

```diff
@@ -1212,8 +1212,8 @@ def _do_dict(solution):
             solution and
             ordered_symbols and
             type(solution) is not dict and
-            type(solution[0]) is dict and
-            all(s in solution[0] for s in symbols)
+            all(type(sol) is dict for sol in solution) and
```

Comment:
Since you're editing this, this should be changed to `isinstance(solution, dict)` (doing `type is` to check types is an antipattern in Python).